exporter/awsemfexporter/metric_translator.go

```diff
@@ -159,10 +168,101 @@ func TranslateOtToCWMetric(rm *pdata.ResourceMetrics, dimensionRollupOption stri
 				continue
 			}
 			cwMetrics := getCWMetrics(&metric, namespace, instrumentationLibName, dimensionRollupOption)
-			cwMetricList = append(cwMetricList, cwMetrics...)
+			cwMetricLists = append(cwMetricLists, cwMetrics...)
+			batchCWMetrics(cwMetrics, groupedCWMetricMap, cwMetricsMap)
 		}
 	}
-	return cwMetricList, totalDroppedMetrics
+	return cwMetricLists, totalDroppedMetrics
+}
+
+func batchCWMetrics(cwMetricList []*CWMetrics, groupedCWMetricMap map[string]*GroupedCWMetric, cwMetricsMap map[string]*CWMetrics) {
+	for _, met := range cwMetricList {
+		if len(met.Measurements[0].Dimensions[0]) > 0 {
```

Comment:
Is this error safe? What about the following edge cases:
1. `met.Measurements` is an empty array 
2. There is more than 1 element in `met.Measurements`
3. There is more than 1 dimension set